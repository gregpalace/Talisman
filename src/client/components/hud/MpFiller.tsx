import React from 'react';

import { MAX_MP } from '../../constants/actionTypes';
import { IMpFiller } from '../../../typings/HUDTypes';

const MpFiller = ({ mp, percentage }: IMpFiller) => {
  return (
    <div
      className='mp-filler'
      style={{ textAlign: 'center', width: `${percentage}%` }}
    >
      MP: {mp >= 20 ? `${mp} / ${MAX_MP}` : ''}
    </div>
  );
};

export default MpFiller;
