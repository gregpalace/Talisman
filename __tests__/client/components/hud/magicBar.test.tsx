import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import MagicBar from '../../../../src/client/components/hud/magicBar';
import MpFiller from '../../../../src/client/components/hud/MpFiller';

describe('React unit tests', () => {
  let wrapper: any;

  const props = {
    mp: 23,
    percentage: 50,
  };

  beforeAll(() => {
    wrapper = shallow(<MagicBar {...props} />);
  });

  describe('<MagicBar />', () => {
    it('should render correctly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render a <div> containing the magicbar', () => {
      expect(wrapper.find('div').prop('className')).toBe('magic-bar');
    });

    it('should render a <MpFiller /> component ', () => {
      const filler = wrapper.find(MpFiller);
      expect(filler).toBeDefined();
      expect(filler.props()).toEqual({ mp: 23, percentage: 50 });
      expect(filler.prop('mp')).toBe(23);
      expect(filler.prop('percentage')).toBe(50);
    });
  });
});
