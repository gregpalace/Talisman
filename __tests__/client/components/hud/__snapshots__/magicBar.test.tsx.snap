// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests <MagicBar /> should render correctly 1`] = `
<div
  className="magic-bar"
>
  <MpFiller
    mp={23}
    percentage={50}
  />
</div>
`;
